core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/internal/SdkClientConfigurationUtil.java

```diff
@@ -55,8 +57,10 @@ public static SdkClientConfiguration.Builder copyOverridesToConfiguration(
 
         // advanced option
         Signer signer = overrides.advancedOption(SdkAdvancedClientOption.SIGNER).orElse(null);
-        builder.option(SdkAdvancedClientOption.SIGNER, signer);
-        builder.option(SdkClientOption.SIGNER_OVERRIDDEN, signer != null);
+        if (signer != null) {
```

Comment:
This logic needed to change like this in order to avoid overriding the client config that sets a default signer if there's no override.